woocommerce/woocommerce-gutenberg-products-block
# Mini Cart Block: render the empty state set on template parts

The PR allows the user to render the empty state set on Mini Cart template parts


Fixes #4674 

### Screenshots

![image](https://user-images.githubusercontent.com/4463174/145204714-81542a78-9d31-41bc-8482-a32a65e95c55.png)
*editor side*

![image](https://user-images.githubusercontent.com/4463174/145204932-7432e07c-afc3-4bae-a812-c6eec445be99.png)
*frontend side*
### Testing

### Manual Testing

How to test the changes in this Pull Request:

Check out this branch. Be sure that you are using `Gutenberg trunk version` and a block theme like `TTY Blocks` and the cart doesn't have any product.



1. Create a post/page with Mini Cart Block.
2. Open the Site Editor.
3. Select `template parts` on the sidebar menu and click on `Mini Cart`.
4. See the switcher in the block toolbar.
5. Set the state `Empty Mini Card`.
6. Edit existent blocks or add new blocks.
7. Go on the post/page saved earlier, click on Mini Cart.
8. Check if you see the blocks you set before in the opened sidebar.


Sometimes, the javascript code has not been downloaded yet, and the content returning to the server is not replaced. Should I open another issue?

https://user-images.githubusercontent.com/4463174/145206702-9a187296-33cb-4990-b225-d5ca38cba592.mp4


My idea is to implement E2E tests too, but I don't want to block @dinhtungdu. Should I create another issue?

## Current problems

I noticed two problems with my implementation:
- the current E2E tests don't work - FIXED
- On the console I see these warnings: - FIXED
```        
The block "woocommerce/empty-mini-cart-contents-block" is registered with an invalid category "woocommerce".
The block "woocommerce/filled-mini-cart-contents-block" is registered with an invalid category "woocommerce".
The block "woocommerce/mini-cart-contents" is registered with an invalid category "woocommerce".
```
But I think that the block registration is correct.




### Changelog

> Mini Cart Block: render the empty state set on Mini Cart template parts
